airflow/providers/oracle/operators/oracle.py

```diff
@@ -93,4 +94,11 @@ def __init__(
     def execute(self, context: Context):
         self.log.info("Executing: %s", self.procedure)
         hook = OracleHook(oracle_conn_id=self.oracle_conn_id)
-        return hook.callproc(self.procedure, autocommit=True, parameters=self.parameters)
+        ti=context.get("task_instance")
+        try:
+            return hook.callproc(self.procedure, autocommit=True, parameters=self.parameters)
+        except oracledb.DatabaseError as e:
+            code, mesg = e.args[0].message[:-1].split(": ", 1)
+            if self.do_xcom_push is True:
```

Comment:
No need to be so explicit

```suggestion
            if self.do_xcom_push:
```